src/lib.rs

```diff
@@ -508,9 +521,23 @@ where
 
                     let wait_time_str = wait_time.to_string();
                     let body = format!("Too many requests, retry in {}s", wait_time_str);
-                    let response = actix_web::HttpResponse::TooManyRequests()
-                        .insert_header((actix_web::http::header::RETRY_AFTER, wait_time_str))
-                        .body(body.clone());
+                    let mut response =
+                        actix_web::HttpResponse::TooManyRequests().body(body.clone());
+
+                    // Add x-ratelimit headers
+                    response.headers_mut().insert(
+                        HeaderName::from_static("x-ratelimit-limit"),
+                        negative.quota().burst_size().get().into(),
+                    );
+                    response.headers_mut().insert(
+                        HeaderName::from_static("x-ratelimit-after"),
+                        wait_time.into(),
+                    );
+                    response.headers_mut().insert(
+                        HeaderName::from_static("x-ratelimit-remaining"),
+                        0u32.into(),
+                    );
```

Comment:
It'd be better to call `headers_mut()` only once and then only `insert()` from the mutable headers variable. Again, this should be optional, not everyone wants to return this headers and this would be a breaking change if enabled by default.

Reply:
Well, after this I will make the user to set the response for not having the key and also the response when the ratelimit is skipped, because it returns a text and in the API we need json

Reply:
> response for not having the key

I will make the user put it when KeyExtractor creates its own, but by default is current, and for the interrogation will make them optional ( add it with method )